virtualhostkeys.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package description
+
+import (
+	"encoding/base64"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+)
+
+type virtualHostKeys struct {
+	Version         int               `yaml:"version"`
+	VirtualHostKeys []*virtualHostKey `yaml:"virtual-host-keys"`
+}
+
+type virtualHostKey struct {
+	ID_      string `yaml:"id"`
+	HostKey_ string `yaml:"host-key"`
+}
+
+// ID implements VirtualHostKey
+func (f *virtualHostKey) ID() string {
+	return f.ID_
+}
+
+// HostKey implements VirtualHostKey
+func (f *virtualHostKey) HostKey() []byte {
+	// Here we are explicitly throwing away any decode error. We encode
+	// a byte array, so we know that the stored host key can be decoded.
+	hostKey, _ := base64.StdEncoding.DecodeString(f.HostKey_)
+	return hostKey
+}
+
+// Validate implements VirtualHostKey.
+func (f *virtualHostKey) Validate() error {
+	if f.ID_ == "" {
+		return errors.NotValidf("empty id")
+	}
+	if len(f.HostKey_) == 0 {
```

Comment:
I guess this can only ever be set using the helper func so if it is not empty, it will be an encoded value so no need to check.

Reply:
Except that if the provided host key slice is empty, this will be an empty string. So I think it's worth checking.